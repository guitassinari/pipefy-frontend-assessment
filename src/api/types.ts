export type Pipe = {
  name: string
  color: string
}