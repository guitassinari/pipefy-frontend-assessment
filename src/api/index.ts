import Queries from './queries'
import * as Types from './types'

export {
  Queries,
  Types
}