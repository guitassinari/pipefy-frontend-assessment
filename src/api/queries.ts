import { gql } from '@apollo/client'

const GET_ORGANIZATION = gql`
  query GetOrganization($id: ID!) {
    organization(id: $id) {
      id
      name
      pipes {
        id
        name
        color
        cards_count
      }
    }
  }
`

const GET_PIPE_CARDS = gql`
  query GetPipeCards($pipeId: ID!, $pageSize: Int, $startAtCursor: String) {
    cards(pipe_id: $pipeId, first: $pageSize, after: $startAtCursor) {
      pageInfo {
        endCursor
        hasNextPage
      }
      edges {
        cursor
        node {
          title
        }
      }
    }
  }
`

export default {
  GET_ORGANIZATION,
  GET_PIPE_CARDS
}