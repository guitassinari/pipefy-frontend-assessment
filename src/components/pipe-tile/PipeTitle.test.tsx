import React from 'react';
import { render } from '@testing-library/react'
import { Types } from 'api'
import PipeTile from '.';

describe('PipeTile', () => {
  describe('when receives a valid pipe', () => {
    const pipe: Types.Pipe = {
      name: 'A pipe'
    }

    it('matches the snapshot', () => {
      const component = render(<PipeTile pipe={pipe} />)
  
      expect(component.asFragment()).toMatchSnapshot()
    })

    it('shows the pipe name', () => {
      const component = render(<PipeTile pipe={pipe} />)
  
      expect(component.getByText(pipe.name)).toBeInTheDocument()
    })
  })
})
