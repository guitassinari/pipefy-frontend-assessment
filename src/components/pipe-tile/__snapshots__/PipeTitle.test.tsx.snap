// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PipeTile when receives a valid pipe matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvtL kfAtnd"
  >
    A pipe
  </div>
</DocumentFragment>
`;
